aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepTest.java

```diff
@@ -27,17 +25,15 @@
     /**
      * The simplest possible timeout step ever.
      */
-    @RandomlyFails("TODO sometimes the whole build just hangs (for longer than the 10s)")
     @Test
     public void basic() throws Exception {
         WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
         p.setDefinition(new CpsFlowDefinition(
-                "node { timeout(time:5, unit:'SECONDS') { sh 'sleep 10'; echo 'NotHere' } }"));
-        WorkflowRun b = assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0));
+                "node { timeout(time:5, unit:'SECONDS') { sleep 10; echo 'NotHere' } }"));
+        WorkflowRun b = assertBuildStatus(Result.ABORTED, p.scheduleBuild2(0));
```

Comment:
why has this changed - I don't see a corresponding TimeoutStep change?

as a user I would expect a timeout to be reported as failure (which matches IIRC standard Jenkins behaviour).
something has failed.